pkg/controllers/streaming/processor_controller.go

```diff
@@ -532,6 +533,18 @@ func (r *ProcessorReconciler) constructDeploymentForProcessor(processor *streami
 		ImagePullPolicy: v1.PullIfNotPresent,
 		Env:             environmentVariables,
 		VolumeMounts:    volumeMounts,
+		ReadinessProbe: &v1.Probe{
+			Handler: v1.Handler{
+				TCPSocket: &v1.TCPSocketAction{
+					Port: intstr.IntOrString{
+						IntVal: 8081,
+					},
+				},
+			},
+			InitialDelaySeconds: 2,
+			TimeoutSeconds:      300,
```

Comment:
300 is a very long timeout. I'd lean towards using the default probe values unless we have a compelling reason to be different. If so, let's document that reason.

Reply:
Ok, using the default for TimeoutSeconds.

Reply:
what about InitialDelaySeconds and PeriodSeconds?

Reply:
I could not find the default value for `InitialDelaySeconds`, but since we are waiting for a JVM to start, I felt 2 seconds was appropriate. Also, `PeriodSeconds` has a default of 10 seconds, which I felt was too big.

Reply:
The readiness probe doesn't need to be super precise since there isn't a service sending traffic to the processor. It's really more of an informational signal to those watching the resource.